src/Storages/VirtualColumnUtils.cpp

```diff
@@ -132,6 +145,15 @@ VirtualColumnsDescription getVirtualsForFileLikeStorage(const ColumnsDescription
     add_virtual("_size", makeNullable(std::make_shared<DataTypeUInt64>()));
     add_virtual("_time", makeNullable(std::make_shared<DataTypeDateTime>()));
 
+    auto map = parseFromPath(path);
+    for (const auto& item : map)
+    {
+        auto type = tryInferDataTypeForSingleField(item.second, settings);
+        if (type == nullptr)
+            type = std::make_shared<DataTypeString>();
+        add_virtual(item.first, std::make_shared<DataTypeLowCardinality>(type));
```

Comment:
We should check `type->canBeInsideLowCardinality()` before making it `LowCardinality`